### PR TITLE
Flexdashboard code review and linking to csv data files

### LINKED ISSUE #68: Create template dashboard for trade statistics using Rshiny and flexdashboard

### LINKED ISSUE #73: Combine plots of the same type into single panels

### PR DESCRIPTION
**Pull request by JosephCrispell**
Hi Rara,

I've spent some time reviewing the code you're written to build the trade statistics dashboard. The current branch covers the following changes:

- Created parameters
    * `useSQL` to check with user whether to read data from SQL server or csv files. Defaults to `FALSE`.
    * `startDate` the start date of month being considered. Defaults to `"2019-01-01"` - as this is the latest full month of data available in the historic data
    * `endDate` the end date of month being considered. Defaults to `"2019-01-31"` - as this is the latest full month of data available in the historic data
- Moved data loading to the start of script and used relative instead of absolute file paths
- Added cell titles (saying what the code in cell does) to each cell
- Added code to either call data from SQL server or local csv files - decided by `useSQL` parameter
- Added comment describing what each package loaded does - **Some more to add**
- Added a favicon - icon that appears at left side on a tab
- Change the interactive bar charts to line charts and combined into single panels (use legend to turn on/off lines)

> Note that to complete the code review I need to following additional files:
> - `country_year_import.csv`
> - `country_year_export.csv`

Some general comments:
- Data files and variables need more informative names
- Not clear what some data files (and SQL tables) are and how they were created
- Variable naming should following consistent structure: `snake_case` or `camelCase` (I'm not the best at this! 😉)
- Avoid hard coding values - throughout script 2019 is hard coded - you can extract this from the `startDate` variable so it will update automatically when year of data changes
- Keep an eye on your tabs in the code
- Don't be afraid to break up code with white space - especially some complex sections
- Add comments above code when it isn't immediately obvious what the code does

All the above are very minor and I think the dashboard is really great and easy to follow. I'm excited to see how it develops! ☺️

Linked to #43 

Closes #68
Closes #72 
Closes #73

### CODE REVIEW

File: R/dashboard.Rmd [hunk 1]
```diff
@@ -1,453 +1,516 @@
...
+library(plotly) # Plotting data interactively
+library(dplyr) # Processing data
+library(openintro)
+library(highcharter)
+library(ggvis)
+library(gganimate)
+library(shiny) # Adding interactive elements to dashboard
+library(RMySQL) # Interacting with local SAL database
+library(lubridate) # Working with dates
+library(openxlsx) # Read excel formatted files
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:34]: Added package descriptions to help understand how we are using package

File: R/dashboard.Rmd [hunk 2]
```diff
@@ -1,453 +1,516 @@
...
+```
+
+
+```{r load data and connect to SQL database}
+
+# Load data
+trades <- read.csv(file.path(repository, "data", "secure", "trades.csv")) # Note that csv files need more informative names
+topten <- read.csv(file.path(repository, "data", "secure", "tab.csv")) # Variables need more informative names
+imp <- read.csv(file.path("data", "secure", "country_year_import.csv")) # I don't have access to this file
+exp <- read.csv(file.path("data", "secure", "country_year_export.csv")) # I don't have access to this file
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:54]: Moved data loading all to start

File: R/dashboard.Rmd [hunk 3]
```diff
@@ -1,453 +1,516 @@
...
+  flexdashboard::flex_dashboard:
+    orientation: rows
+    vertical_layout: scroll
+    social: [ "twitter", "facebook", "menu"]
+    source_code: embed
+    runtime: shiny
+    favicon: favicon.ico
+    
+params:
+    useSQL: FALSE # Select whether to load data from SQL server
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:13]: Parameter to check whether to load data from SQL or not

File: R/dashboard.Rmd [hunk 4]
```diff
@@ -1,453 +1,516 @@
...
+    startDate: "2019-01-01" # Start date for latest month (YYYY-dd-mm)
+    endDate: "2019-01-31" # End date for latest month (YYYY-dd-mm)
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:15]: Dates defining the month of interest

File: R/dashboard.Rmd [hunk 5]
```diff
@@ -1,453 +1,516 @@
...
+exp <- read.csv(file.path("data", "secure", "country_year_export.csv")) # I don't have access to this file
+
+# Connect to local SQL database if requested
+mydb <- NULL
+historicImports <- NULL
+historicExports <- NULL
+historicImportsLatestMonth <- NULL
+historicExportsLatestMonth <- NULL
+principleCOmmodities <- NULL
+if(params$useSQL){
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:63]: Using the `params$useSQL` in code chunk to load data either from SQL or from local files

File: R/dashboard.Rmd [hunk 6]
```diff
@@ -1,453 +1,516 @@
...
+
+Dashboard
+=====================================
+
+Row
+-------------------------------------
+
+### Goods Import/Export Analysis
+
+```{r month value box}
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:98]: Add R cell labels throughout

File: R/dashboard.Rmd [hunk 7]
```diff
@@ -1,453 +1,516 @@
...
+valueBox(import_cur,
+         icon = 'fa-arrow-left',
+         color = "blue")
+```
+
+### Total Current Month Export
+
+```{r total export value box}
+# Calculate the total export for latest month
+export_cur <- NULL
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:126]: Using snake_case here but elsewhere using camelCase - either is fine but try and be consistent

File: R/dashboard.Rmd [hunk 8]
```diff
@@ -1,453 +1,516 @@
...
+            y = ~Import,
+            color = "red",
+            name = "Import") %>%
+  
+  add_trace(x = ~Year,
+            y = ~Balance,
+            color = "green",
+            name = "Balance") %>%
+  
+  layout(xaxis = list(title = "Year"), yaxis = list(title = 'Value'))
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:242]: Changed to line chart and combined into single panel - user can select line using legend

File: R/dashboard.Rmd [hunk 9]
```diff
@@ -1,453 +1,516 @@
...
+    group_by(Year) %>%
+    summarise("Totex"=sum(Value, na.rm=TRUE)/1000000, .groups="keep")
+  
+  goodsImport <- historicImports %>%
+    filter(Procedure %in% c(4000, 4021, 7100)) %>%
+    group_by(Year) %>%
+    summarise("Totim"=sum(Value, na.rm=TRUE)/1000000, .groups="keep")
+}
+
+trade_combine_summary <- merge(goodsExport, goodsImport, by="Year")
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:304]: Mixture of camelCase and snake_case here

File: R/dashboard.Rmd [hunk 10]
```diff
@@ -1,453 +1,516 @@
...
+```
+
+Row
+--------------------------------
+### Trade Activities with 10 top countries
+
+
+```{r animated graphic for top 10 countries by year for imports}
+
+imp %>%
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:315]: Couldn't test this code as couldn't access necessary files

File: R/dashboard.Rmd [hunk 11]
```diff
@@ -1,453 +1,516 @@
...
+  
+  transition_states(Year, transition_length = 4, state_length = 1, wrap = FALSE) +
+  
+  view_follow(fixed_x = TRUE)  +
+  
+  labs(title = 'Export per Year : {closest_state}',  
+       subtitle  =  "Top 10 Countries Vanuatu has exported to",
+       caption  = "Export Value in Billions of Vatu | Data Source: Asycuda") 
+
+animate(anim_exp, 200)
```
Comment (JosephCrispell) [inline R/dashboard.Rmd:430]: Added some white space to break code up. Also fixed some of the tabbing

File: R/dashboard.Rmd (hunk 2 above)
Comment (JosephCrispell) [inline R/dashboard.Rmd:54]: variable and file names could be more informative. Also not clear how these data files were created.

**Comment by JosephCrispell** [comment]
Hi @trara538,

I've now got the dashboard working well on my laptop (I had to fix a few errors I had created and point to the data files you shared). It would great if you can pull the latest version and test the dashboard on your computer - let me know how you get on.

Once the dashboard is working ok on your computer we can spend some time reviewing the changes and then merge the branch.

**Review by trara538 — APPROVED** [review]
Did some work and we can continue from here on

**Pull request merged**